File: 0942-di-string-match/0942-di-string-match.java
class Solution {
    public int[] diStringMatch(String s) {
        int n=s.length();
        int left=0;
        int right=n;
        int arr[]=new int[n+1];
        int ind=0;

        for(char c:s.toCharArray()){
            if(c=='I'){
                arr[ind]=left;
                left++;
            }
            else{
                arr[ind]=right;
                right--;
            }
            ind++;
        }
        if(left<=right){
            arr[ind]=left;
        }
        return arr;
    }
}

File: 0942-di-string-match/0942-di-string-match.py
class Solution(object):
    def diStringMatch(self, s):
        """
        :type s: str
        :rtype: List[int]
        """
        n=len(s)
        arr=[0]*(n+1)
        left=0
        right=n
        ind=0
        for i in s:
            if i=='I':
                arr[ind]=left
                left+=1
            else:
                arr[ind]=right
                right-=1
            ind+=1
        if left<=right: arr[ind]=left
        return arr